src/main/java/com/gluonhq/substrate/config/ConfigResolver.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.substrate.config;
+
+import com.gluonhq.substrate.model.ClassPath;
+import com.gluonhq.substrate.util.FileOps;
+import com.gluonhq.substrate.util.Logger;
+import com.gluonhq.substrate.util.Strings;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import static com.gluonhq.substrate.Constants.USER_INIT_BUILD_TIME_ARCHOS_FILE;
+import static com.gluonhq.substrate.Constants.USER_INIT_BUILD_TIME_FILE;
+import static com.gluonhq.substrate.Constants.META_INF_SUBSTRATE_CONFIG;
+import static com.gluonhq.substrate.Constants.USER_JNI_ARCHOS_FILE;
+import static com.gluonhq.substrate.Constants.USER_JNI_FILE;
+import static com.gluonhq.substrate.Constants.USER_REFLECTION_ARCHOS_FILE;
+import static com.gluonhq.substrate.Constants.USER_REFLECTION_FILE;
+
+/**
+ * Helper class that helps scanning jars in the classpath looking for
+ * files in META-INF/substrate/config that contain information that has
+ * to be added to the config files and native image command line flags
+ */
+public class ConfigResolver {
+
+    private final List<File> jars;
+
+    public ConfigResolver(String classpath) {
+        this.jars = new ClassPath(classpath)
+                .filter(s -> s.endsWith(".jar") && !s.contains("javafx-")).stream()
+                .map(File::new)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Walks through the jars in the classpath, excluding the JavaFX ones,
+     * and looks for META-INF/substrate/config/initbuildtime or
+     * META-INF/substrate/config/initbuildtime-${archos} files.
+     *
+     * The method will return a list of class names from all the files found
+     *
+     * @param archOs a string with the arch and os
+     * @return a list of classes that should be initialized at build time
+     * @throws IOException
+     */
+    public List<String> getUserInitBuildTimeList(String archOs) throws IOException {
+        Logger.logDebug("Scanning for init build time files");
```

Comment:
Would be nice to check the `archOS` for null. Previous method does not need this as `Classpath` does it already.